### ISSUE TITLE
music (getid3) don't work when encryption is enabled

### ISSUE THREAD

**Issue by burnersk** [owncloud/music#153]
Using ownCloud 6.0.1 `music` (0.1.7) don't work well when the `encryption` extension is enabled. `music` will decrypt the files for playing but not for getid3 to extract tags. After disableing `encryption` (and wait a few hours) and truncate the `music` database as mentioned in debugging `music` all mp3 tags was extracted correctly. Here are some common errors I got with enabled `encryption`:

```
extracted metadata - artist: ownCloud unknown artist, album: ownCloud unknown album, title: test.mp3, track#: , year: , mimetype: audio\/mpeg, fileId: 
openssl_decrypt(): Failed to base64 decode the input at \/srv\/owncloud\/apps\/files_encryption\/lib\/crypt.php#240
getID3 error message: error parsing \"APIC\" (196 bytes into the ID3v2.3 tag). (ERROR: $frame_size (36084) > strlen($framedata) (null)).
getID3 error message: Caught exception: Encryption library: Decryption (symmetric) of content failed
getID3 error message: Could not find valid MPEG synch before end of file
getID3 error occured
```

**Comment by stefarossi** [comment]
I'm running OwnCloud 8.0.2 and I just ran into this bug, so I can see that this is still is a problem.
Is anybody working on it?

**Comment by stefarossi** [comment]
I wrote a very ugly workaround for fixing this. I basically fully decrypt the file before passing it to getID3. There must be a better way, I'll keep investigating. If the "seasoned developers" can give me a tip on what's the best way to fix this, I can help with the coding.

Here's the workaround. In 'utility/extractorgetid3.php' I changed this block:
```
public function extract($path) {
	$metadata = $this->getID3->analyze($path);

```
to
```
public function extract($path) {
	// WORKAROUND
	$tempfile = tempnam(sys_get_temp_dir(), 'ocmusic-');
	$read_data = file_get_contents($path);	
	file_put_contents($tempfile, $read_data);

	$metadata = $this->getID3->analyze($tempfile);

	unlink($tempfile);
```

There's also the problem that, if encryption is enabled, the browser has to fully load the file before it can start playing it. I have no idea on how to fix that.

**Comment by paulijar** [comment]
I tested encryption with Nextcloud 13.0.0-alpha, Default encryption module, Music app 0.5.4, and Tomahawk 0.8.4 for Windows. All the album and artists names and other metadata were extracted and shown correctly. My best guess is that this has got fixed by the commit 330fc142a27d44. Closing.
